File: netcore-api/Controllers/ValuesController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;

namespace netcore_api.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class ValuesController : ControllerBase
    {
        string fileName = "/home/fcrozetta/projetos/beta/nomes.txt";
        // GET api/values
        [HttpGet]
        public ActionResult<IEnumerable<string>> Get()
        {
            System.IO.StreamReader sr = new System.IO.StreamReader(fileName);
            return sr.ReadToEnd().ToString().Split('\n');
        }


        // POST api/values
        [HttpPost]
        public void Post([FromBody] Dictionary<string,string> req)
        {
            System.IO.File.AppendAllText(fileName,$"{req["name"]}{Environment.NewLine}");
        }

    }
}

File: nest-api/src/values/values.controller.ts
import { Controller, Get, Req, Post, Param, Body } from '@nestjs/common';
import {Request } from 'express'
var fs = require('fs');

@Controller('values')
export class ValuesController {
    nameFile = "/home/fcrozetta/projetos/beta/nomes.txt"
    @Get()
    getValues(): string {
        let names = fs.readFileSync(this.nameFile).toString().split('\n')
        return names
    }
    
    @Post()
    postValue(@Body() req ): string{
        fs.appendFile(this.nameFile,req['name']+"\n",(err)=>{
            if (err) {
                console.log("ERRO ao gravar arquivo de nomes")
                throw err;
            }
        })
        return req.name
    }

    
}
